### PR TITLE
Fixed some problems in matrices

### LINKED ISSUE #9052: `trace(2*A) != 2*Trace(A)` because LHS still has an MatMul
````
n = Symbol('n', integer=True)
A = MatrixSymbol('A', n, n)
trace(2*A) == 2*Trace(A)    # False
srepr(trace(2*A))   # note 2*Trace(MatMul(A))
````

### PR DESCRIPTION
**Pull request by kevinventullo**
Fixes #9052 and #9053.

For 9052, I added .doit() to _eval_trace.

For 9053, I modified merge_explicit in MatMul to take scalars into account.

### CODE REVIEW

File: sympy/matrices/expressions/matadd.py [hunk 1]
```diff
@@ -50,8 +50,8 @@ def _eval_adjoint(self):
         return MatAdd(*[adjoint(arg) for arg in self.args]).doit()
 
     def _eval_trace(self):
-        from trace import Trace
-        return MatAdd(*[Trace(arg) for arg in self.args]).doit()
+        from .trace import trace
+        return Add(*[trace(arg) for arg in self.args]).doit()
```
Comment (cbm755) [inline sympy/matrices/expressions/matadd.py]: this bit looks the same (?) as #9031.

**Comment by cbm755** [comment]
this will need some tests.  Perhaps appropriate ones are XFAILd in #9031?  One way is to base this on top of #9031..., but it needs rebased first.

**Comment by cbm755** [comment]
@kevinventullo ok I've squashed #9031 to give #9063.  Maybe rebase on #9063?   Or hopefully we'll get that merged and you can rebase on master.  Note it has XFAIL tests that should pass with your changes.

Does this need any other tests?

File: sympy/matrices/expressions/matadd.py (hunk 1 above)
Comment (kevinventullo) [inline sympy/matrices/expressions/matadd.py]: Yes, I ended up throwing that in since I did something similar in MatMul.

**Comment by kevinventullo** [comment]
I went through #9063 and I believe this PR passes all the XFAIL tests except 
```
q = MatAdd(X, 2*X)
assert Trace(q).doit(deep=False).arg == q
```
and those in `test_Trace_MutableMatrix_plus()`. I could throw in a few more tests for good measure

**Comment by cbm755** [comment]
@kevinventullo if you're able to review #9063, we might get these in a bit faster...

**Comment by kevinventullo** [comment]
@cbm755 Sure thing, didn't realize I could be part of the review process. Was +1'ing it enough?

**Comment by cbm755** [comment]
We welcome anyone to participate in reviewing: there is so much to do.

See also https://github.com/sympy/sympy/wiki/Development-workflow

**Comment by cbm755** [comment]
*After* #9063 is merged, you'll probably need to merge master onto your branch, see dev URL.

**Comment by cbm755** [comment]
some test failures?  and/or do you need to remove some XFAIL's from other tests?

**Comment by kevinventullo** [comment]
I appended ".doit()" to the end of the RHS of the two failed tests. Is that all right?

**Comment by cbm755** [comment]
Maybe, I'll review when I get a chance.  But there should still be some XFAIL tests that now pass right?  This PR should remove the "XFAIL" line from those

**Comment by certik** [comment]
This looks good to me. @cbm755 which @XFAIL should be removed? Or we can merge it and remove XFAIL later.

**Comment by cbm755** [comment]
Ok I'm +1 too.

I can do another PR for the XFAIL and Authors/aboutus.  I'll close the issues there too.

I will merge in 24 hours, unless @certik does it sooner.

**Comment by certik** [comment]
Thanks @cbm755.

**Pull request merged**